jwst/ramp_fitting/ramp_fit_step.py

```diff
@@ -456,6 +458,30 @@ def process(self, step_input):
 
             int_times = result.int_times
 
+            # Set the DO_NOT_USE bit in the groupdq values for groups before firstgroup
```

Comment:
I recommend moving this section of code to a function and calling that function conditioned on `if firstgroup is not None or lastgroup is not None:`.